pkg/model/rhsso_deployment.go

```diff
@@ -121,10 +121,33 @@ func getRHSSOEnv(cr *v1alpha1.Keycloak, dbSecret *v1.Secret) []v1.EnvVar {
 		env = MergeEnvs(cr.Spec.KeycloakDeploymentSpec.Experimental.Env, env)
 	}
 
+	env = RHSSOSslEnvVariables(dbSecret, env)
+
 	return env
 }
 
-func RHSSODeployment(cr *v1alpha1.Keycloak, dbSecret *v1.Secret) *v13.StatefulSet {
+func RHSSOSslEnvVariables(dbSecret *v1.Secret, env []v1.EnvVar) []v1.EnvVar {
+	if dbSecret != nil {
+		sslMode := string(dbSecret.Data[DatabaseSecretSslModeProperty])
+
+		if sslMode != "" {
+			// append env variable
+			env = append(env,
+				v1.EnvVar{
+					Name:  RhssoDatabaseXAConnectionParamsProperty + "_sslMode",
+					Value: sslMode,
+				},
+				v1.EnvVar{
+					Name:  RhssoDatabaseNONXAConnectionParamsProperty + "_sslmode",
```

Comment:
Based on the discussion with @iankko, this needs to be:
```suggestion
					Name:  RhssoDatabaseNONXAConnectionParamsProperty + "_sslMode",
```